avogadro/qtplugins/atompairbonds/atompairbonds.cpp

```diff
@@ -0,0 +1,192 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#include "atompairbonds.h"
+
+#include <avogadro/core/array.h>
+#include <avogadro/core/bond.h>
+#include <avogadro/core/elements.h>
+#include <avogadro/core/neighborperceiver.h>
+#include <avogadro/qtgui/molecule.h>
+#include <avogadro/rendering/dashedlinegeometry.h>
+#include <avogadro/rendering/geometrynode.h>
+#include <avogadro/rendering/groupnode.h>
+
+#include <QtCore/QSettings>
+#include <QtGui/QColor>
+#include <QtWidgets/QDoubleSpinBox>
+#include <QtWidgets/QFormLayout>
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QWidget>
+
+namespace Avogadro {
+namespace QtPlugins {
+
+using Core::Array;
+using Core::Bond;
+using Core::NeighborPerceiver;
+using QtGui::Molecule;
+using QtGui::PluginLayerManager;
+using Rendering::DashedLineGeometry;
+using Rendering::GeometryNode;
+using Rendering::GroupNode;
+
+AtomPairBonds::AtomPairBonds(QObject *p) : ScenePlugin(p)
+{
+  m_layerManager = PluginLayerManager(m_name);
+  
+  QSettings settings;
+  m_angleToleranceDegrees = settings.value("atomPairBonds/angleTolerance", 20.0).toDouble();
+  m_maximumDistance = settings.value("atomPairBonds/maximumDistance", 2.0).toDouble();
+  QColor hydrogenBColor = settings.value("atomPairBonds/lineColor0", QColor(64, 192, 255)).value<QColor>();
+  m_lineColors = {
+    Vector3ub(hydrogenBColor.red(), hydrogenBColor.green(), hydrogenBColor.blue())
+  };
+  m_lineWidths = {
+    settings.value("atomPairBonds/lineWidth0", 2).toInt()
+  };
+}
+
+AtomPairBonds::~AtomPairBonds() {}
+
+enum InteractionTypes {
+  NONE = -1,
+  HYDROGEN_BOND = 0
+};
+
+static enum InteractionTypes getInteractionType(const Molecule &molecule, Index i) {
+  unsigned char inum = molecule.atomicNumber(i);
+  switch (inum) {
+    case 1: // hydrogen bond
+      for (const Bond *b : molecule.bonds(i)) {
+        Index j = (b->atom1().index() == i ? b->atom2() : b->atom1()).index();
```

Comment:
I tend to agree with the MISRA rule here.. I don't really like the ? operator -- it's hard to read. Can you just change this to an if/else syntax?